File: services/campus-market-parent/cm-notify/src/main/java/com/campus/market/notify/entity/Message.java
package com.campus.market.notify.entity;

import com.campus.market.notify.enums.MessageType;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

/**
 * 消息实体
 */
@Entity
@Table(name = "biz_message")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "user_id", nullable = false)
    private Long userId;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private MessageType type;
    
    @Column(nullable = false, length = 200)
    private String title;
    
    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;
    
    @Column(name = "is_read", nullable = false)
    private Boolean isRead = false;
    
    @Column(name = "related_id")
    private Long relatedId;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: frontend/src/types/message.ts
/* ============================================================
   消息通知 DTO
   依据 docs/赏金布_API对接过渡设计.md 5.10
   ============================================================ */
import type { MessageType } from './enums'

/** 消息视图对象 */
export interface MessageVO {
  id: number
  userId: number
  type: MessageType
  title: string
  content: string
  isRead: boolean
  relatedId?: number
  createdAt: string
}
